src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -263,7 +263,7 @@ public void trimToNewStartPosition( final int newStart ) {
             }
 
             // Record our new start boundary
-            cachedInterval = new SimpleInterval(cachedInterval.getContig(), newStart, cachedInterval.getEnd());
+            cachedInterval = new SimpleInterval(cachedInterval);
```

Comment:
That's because it's not a correctness bug, it's an efficiency bug (stale entries in the cache before the current location were not getting evicted properly)